src/adapters/http/server.ts

```diff
@@ -93,7 +93,7 @@ class HttpAdapter extends Adapter {
         this.httpResponses.set(this.serverName, res)
         let { pathname } = new URL(req.url, serverUrl)
         pathname = pathname.startsWith('/') ? pathname.substring(1) : pathname
-        if (!this.parsedAsyncAPI.channel(pathname)) {
+        if (!this.parsedAsyncAPI.channels().get(pathname)) {
```

Comment:
Same as above. The channel name is not necessarily the address. You should find the channel that has the `pathname` address.